setup.tpl

```diff
@@ -88,4 +89,4 @@ vault write auth/azure/login \
   vm_name="`hostname`"
```

Comment:
`hostname` doesn't work, as `vm_name` should be the vm id. This results in, for example, the resource
```
Microsoft.Compute/virtualMachines/vault-vmss000002
```
while it should be (?)
```
Microsoft.Compute/virtualMachines/2
```

Error message:
```
Error writing data to auth/azure/login: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/auth/azure/login
Code: 500. Errors:

* unable to retrieve virtual machine metadata: compute.VirtualMachinesClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource
'Microsoft.Compute/virtualMachines/vault-vmss000002' under resource group 'vault-rg' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
```